### PR TITLE
Improving examples

### LINKED ISSUE #880: Move examples out of crates/matrix-sdk
**Is your feature request related to a problem? Please describe.**

It's somewhat annoying to have to navigate to `crates/matrix-sdk/examples` instead of just `examples`, plus this nesting makes the examples much harder to discover. Also, they don't have individual lists of dependencies, making it harder to use them as a starting point for new projects.

**Describe the solution you'd like**

Let's make the examples their own crates, same as in [Ruma](https://github.com/ruma/ruma/tree/main/examples), [axum](https://github.com/tokio-rs/axum/tree/main/examples) and others.

We should think about whether it makes sense to have them be part of the regular Cargo workspace (like in Ruma) or as a separate Cargo workspace (like in axum). Context for why this is separated in axum: https://github.com/tokio-rs/axum/pull/978

### PR DESCRIPTION
**Pull request by gnunicorn**
- [x] move examples into separate folder, fixes #880
- [x] create getting-started example to use for workshops
- [x] create custom-events example to explain devs how to use custom events
- [x] add Readme for Examples

### CODE REVIEW

File: crates/matrix-sdk/src/client/builder.rs [hunk 1]
```diff
@@ -132,7 +132,7 @@ impl ClientBuilder {
     ///
     /// This is a shorthand for
     /// <code>.[store_config](Self::store_config)([matrix_sdk_indexeddb]::[make_store_config](matrix_sdk_indexeddb::make_store_config)(path, passphrase).await?)</code>.
-    #[cfg(feature = "indexeddb")]
+    #[cfg(all(feature = "indexeddb", target_arch = "wasm32"))]
```
Comment (jplatte) [inline crates/matrix-sdk/src/client/builder.rs:135]: This will make them disappear from docs. What was clippy complaining about?

Comment (gnunicorn) [reply crates/matrix-sdk/src/client/builder.rs:135]: that returning `OpenStoreError` doesn't exist, which is only exposed by `indexeddb` when under `wasm32` arch ... I honestly don't understand why that would have ever passed...

Comment (jplatte) [reply crates/matrix-sdk/src/client/builder.rs:135]: I see. I think this should fix it: https://github.com/matrix-org/matrix-rust-sdk/pull/910

File: examples/Readme.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Matrix SDK Examples
```
Comment (jplatte) [inline examples/Readme.md:1]: readme files usually have an all-uppercase filename (except the extension). For consistency I think we should do the same thing here.

**Review by jplatte — COMMENTED** [review]
One more small comment, otherwise I'm just waiting on the indexeddb stuff being reverted to approve.

**Review by jplatte — APPROVED** [review]

**Comment by jplatte** [comment]
This now just needs a rebase, right?

File: examples/custom_events/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        },
+        OwnedEventId,
+    },
+    Client,
+};
+use serde::{Deserialize, Serialize};
+use tokio::time::{sleep, Duration};
+
+// We use ruma to define our custom events. Just declare the events content
+// by deriving from `EventContent` adn define `ruma_events` for the metadata
```
Comment (jplatte) [inline examples/custom_events/src/main.rs:34]:
```suggestion
// by deriving from `EventContent` and define `ruma_events` for the metadata
```

File: examples/custom_events/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+#[ruma_event(type = "rs.matrix-sdk.example.ack", kind = MessageLike)]
+pub struct AckContent {
+    // the event ID of the ping.
+    ping_id: OwnedEventId,
+}
+
+/// The final events can be redacted, hence us defining only the specialized
+/// content (and not room_id and event_id and such). Wrapping the content with
+/// `SyncMessageLikeEvent` props them up to become a full event.
+pub type PingEvent = SyncMessageLikeEvent<PingContent>;
```
Comment (jplatte) [inline examples/custom_events/src/main.rs:50]: The `EventContent` derive generates type aliases, I think those should be used instead. If `PingContent` was named `PingEventContent` would exist already (but `SyncPingEvent` would be the type alias for this).

File: examples/custom_events/src/main.rs [hunk 5]
```diff
@@ -31,24 +31,24 @@ use serde::{Deserialize, Serialize};
...
     // the event ID of the ping.
     ping_id: OwnedEventId,
 }
 
-/// The final events can be redacted, hence us defining only the specialized
-/// content (and not room_id and event_id and such). Wrapping the content with
-/// `SyncMessageLikeEvent` props them up to become a full event.
-pub type PingEvent = SyncMessageLikeEvent<PingContent>;
-pub type AckEvent = SyncMessageLikeEvent<AckContent>;
+// !!Important!!: Deriving `EventContent` generates a few types and aliases,
```
Comment (jplatte) [inline examples/custom_events/src/main.rs:47]: I'm not sure what `!!Important!!` accomplishes…

**Review by jplatte — APPROVED** [review]
Thanks for adding the custom event example!

**Pull request merged**